tests/test_p2.py

```diff
@@ -1,39 +1,113 @@
     G.add_edge(3, 6)
 
+    return G
+
+def test_p2_should_be_applied_full_proper_graph(initial_full_graph):
+    G = initial_full_graph
+
     assert True == P2.apply(G)
 
-    assert [(1, {'label': 'el', 'pos': (0.5, 0.5), 'layer': 0}),
-            (2, {'label': 'i', 'pos': (0.5, 0.5), 'layer': 1}),
-            (3, {'label': 'E', 'pos': (0, 0), 'layer': 1}),
-            (4, {'label': 'E', 'pos': (1, 0), 'layer': 1}),
-            (5, {'label': 'E', 'pos': (0, 1), 'layer': 1}),
-            (6, {'label': 'E', 'pos': (1, 1), 'layer': 1}),
-            (7, {'label': 'I', 'pos': (0.25, 0.5), 'layer': 2}),
-            (8, {'label': 'I', 'pos': (0.75, 0.5), 'layer': 2}),
-            (9, {'label': 'E', 'pos': (0, 0), 'layer': 2}),
-            (10, {'label': 'E', 'pos': (0.5, 0.0), 'layer': 2}),
-            (11, {'label': 'E', 'pos': (1, 0), 'layer': 2}),
-            (12, {'label': 'E', 'pos': (0, 1), 'layer': 2}),
-            (13, {'label': 'E', 'pos': (0.5, 1.0), 'layer': 2}),
-            (14, {'label': 'E', 'pos': (1, 1), 'layer': 2})] == list(G.nodes(data=True))
-
-    assert {(1, 2), (2, 3), (2, 4), (2, 5), (2, 6), (2, 8), (2, 7), (3, 4), (3, 6), (4, 5),
-            (5, 6), (7, 9), (7, 10), (7, 12), (7, 13), (8, 10), (8, 11), (8, 13), (8, 14),
-            (9, 10), (9, 12), (10, 11), (10, 13), (11, 14), (12, 13), (13, 14)} == set(G.edges)
+    assert [
+     (1, {'label': 'el', 'pos': (0.5, 0.5), 'layer': 0}),
+     (2, {'label': 'i', 'pos': (0.5, 0.5), 'layer': 1}),
+     (3, {'label': 'E', 'pos': (0, 0), 'layer': 1}),
+     (4, {'label': 'E', 'pos': (1, 0), 'layer': 1}),
+     (5, {'label': 'E', 'pos': (1, 1), 'layer': 1}),
+     (6, {'label': 'E', 'pos': (0, 1), 'layer': 1}), 
+     (7, {'label': 'I', 'pos': (0.25, 0.5), 'layer': 2}), 
+     (8, {'label': 'I', 'pos': (0.75, 0.5), 'layer': 2}), 
+     (9, {'label': 'E', 'pos': (0, 0), 'layer': 2}), 
+     (10, {'label': 'E', 'pos': (0.5, 0.0), 'layer': 2}), 
+     (11, {'label': 'E', 'pos': (1, 0), 'layer': 2}), 
+     (12, {'label': 'E', 'pos': (0, 1), 'layer': 2}), 
+     (13, {'label': 'E', 'pos': (0.5, 1.0), 'layer': 2}), 
+     (14, {'label': 'E', 'pos': (1, 1), 'layer': 2})] == list(G.nodes(data=True))
+
+    assert {
+    (3, 4), (12, 13), (2, 5), (11, 14), (2, 8), (13, 14), (7, 10), (7, 13), (4, 5), 
+    (5, 6), (3, 6), (8, 11), (9, 10), (2, 4), (1, 2), (8, 14), (10, 11), (2, 7), 
+    (7, 9), (7, 12), (9, 12), (8, 10), (10, 13), (2, 3), (8, 13), (2, 6)} == set(G.edges)
+
+
+def test_p2_should_be_applied_without_edge_to_initial_node(initial_full_graph):
+    G = initial_full_graph
+    
+    G.remove_edge(1, 2)
+
+    assert True == P2.apply(G)
+
+
+def test_p2_should_be_applied_with_wrong_label_in_initial_node(initial_full_graph):
+    G = initial_full_graph
+    
+    G.nodes[1]['label'] = 'WRONG LABEL'
+
+    assert True == P2.apply(G)
+
+def test_p2_should_not_be_applied_because_of_missing_outer_edge(initial_full_graph):
```

Comment:
maybe `for` isntead of `because_of`. It is shorter and I think it means the same. And also some tests have `with`.

Reply:
so what?

Reply:
to unify tests names

Reply:
yeah they have 'with' because they start with 'should_be_applied' , others that have 'because_of" start with 'should_not_be_applied`